tools/tdx-measure/tdvf/tdvf.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// This package implements parsing of firmware files following the Intel® TDX
+// Virtual Firmware Design Guide.
+package tdvf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/contrast/tdx-measure/mrtd"
+)
+
+var (
+	tableFooterGUID       = []byte{0xde, 0x82, 0xb5, 0x96, 0xb2, 0x1f, 0xf7, 0x45, 0xba, 0xea, 0xa3, 0x66, 0xc5, 0x5a, 0x08, 0x2d}
+	tdxMetadataOffsetGUID = []byte{0x35, 0x65, 0x7a, 0xe4, 0x4a, 0x98, 0x98, 0x47, 0x86, 0x5e, 0x46, 0x85, 0xa7, 0xbf, 0x8e, 0xc2}
+	metadataSignature     = "TDVF"
+)
+
+// Find the offset for the TDVF metadata as described in Intel® TDX Virtual
+// Firmware Design Guide, 11.1 TDVF Metadata Location.
+func findTdvfMetadataOffset(firmare []byte) (uint32, error) {
+	footerGUID := firmare[len(firmare)-0x30:][:0x10]
```

Comment:
Why are some indices in this file base16 and some base10?

Reply:
I don't think there's a good reason. The Intel spec uses base 10 for the TDVF descriptor/sections and a single base 16 number for describing the metadata offset. IIRC there's another SEV spec that goes into a bit more detail about the metadata offset, but I can't find it right now, though I suspect that maybe that spec uses base 16.

I changed all offsets to be in base 10.